apisix/plugins/basic-auth.lua

```diff
@@ -161,6 +170,11 @@ function _M.rewrite(conf, ctx)
         return 401, { message = "Password is error" }
     end
 
+    -- 5. hide `Authentication` request header if `hide_auth_header` is `true`
+    if conf.hide_auth_header == true then
+        core.request.set_header(ctx, "Authentication", "")
+    end
+
```

Comment:
The original issue seems to avoid to send the header to upstream.

Reply:
Yes, use `core.request.set_header(ctx, "Authentication", nil)` can avoid.